### DISCUSSION TITLE
Cpu time study

### DISCUSSION THREAD

**Discussion by PedroMPagani** [xetorthio/jedis#3213] [Q&A] [answered]
Is there any specific reason why on this profiler
![image](https://user-images.githubusercontent.com/56891617/203197551-789d99fa-7fe7-43aa-a990-b0e6a63ad393.png)
it points that specific method as 0.55ms and the actual .flush as 0.05

**Comment by sazzad16** [comment, marked as answer]
It represents everything that is not handled in your Java application.

It includes actual network time and server processing time:
1. Data reaching to Redis server host over network
2. (Maybe) Waiting in processing queue
3. Processing by Redis server
4. Reply back from Redis server over network

**Reply by PedroMPagani** [reply]
> It represents everything that is not handled in your Java application.
> 
> It includes actual network time and server processing time:
> 1. Data reaching to Redis server host over network
> 2. (Maybe) Waiting in processing queue
> 3. Processing by Redis server
> 4. Reply back from Redis server over network

I would guess the reply back from redis is the biggest CPU consumer in here? Alongside with the latency of coming back?

**Reply by sazzad16** [reply]
I ain't saying that. I can only say it's the combination of all of them.

**Comment by zeekling** [comment]
costs of .flush : The cost send command to redis server
costs of .readProtocolWithCheckingBroken : The cost redis server process command + The cost client read result.

The max costs is not sure, The cost redis server process command or The cost client read result.
